proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,601 @@
+
+* Parse-friendly output. Other package managers like [NPM already have it](https://docs.npmjs.com/cli/v7/commands/npm-ls) (`npm ls --parseable` and `npm ls --json`).
+* Useful for CI/CD auditing(compliance, security ..)
+  * Produce Software Bill of Material (SBOM) (compliance, historic keeping)
+  * Enhancing Software Supply Chain Security
+    * Check any vulnerable packages
+    * Check any deprecated packages
+    * Check any outdated packages
+  * Check license compliance
+  * Resolve dependency issues (detect duplicate packages, newly introduced dependencies, dependency removals, etc.)
+  * Making the output machine-readable unlocks additional tooling and automation scenarios in CI/CD pipeline above scenarios.
+
+## Who are the customers
+
+Anyone (government/private enterprises, security experts, individual contributors) who wants to consume `dotnet list package` output for auditing tool or historic keeping, CI/CD orchestrating.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+#### `--format` option
+
+Ability to use new `--format` option for all `dotnet list package` commands to ensure formatted(JSON etc) output is emitted to the console.
+
+```dotnetcli
+dotnet list [<PROJECT>|<SOLUTION>] package [--config <SOURCE>]
+    [--deprecated]
+    [--framework <FRAMEWORK>] [--highest-minor] [--highest-patch]
+    [--include-prerelease] [--include-transitive] [--interactive]
+    [--outdated] [--source <SOURCE>] [-v|--verbosity <LEVEL>]
+    [--vulnerable]
+    [--format <FORMAT>]
+
+dotnet list package -h|--help
+```
+
+`<FORMAT>` - Allowed values as part of spec is `json`. (In the future `parseable`, `csv`, `yaml`, `xml` could be candidates.)
+
+#### `> dotnet list package`
+
+```dotnetcli
+Project 'MyProjectA' has the following package references
+   [netcoreapp3.1]:
+   Top-level Package                      Requested             Resolved
+   > Microsoft.Extensions.Primitives      [1.0.0, 5.0.0]        1.0.0
+   > NuGet.Commands                       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib                         [1.1.2, 2.0.0)        1.1.2
+
+Project 'MyProjectB' has the following package references
+   [netcoreapp3.1]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+
+   [net5.0]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+```
+
+#### `> dotnet list package --format json`
+
+```json
+{
+  "version": 1,
+  "MyProjectA": [
```

Comment:
What happens if I name my project `version`? Perhaps nest projects' information in a `project` node, as otherwise this may make it difficult for us to add additional properties in the future

Reply:
Great point. Just addressed. Please check.